src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -189,12 +193,20 @@ def _to_df(self, agg_exprs: List[Expression]) -> DataFrame:
             raise TypeError(f"Wrong group by type {self._group_type}")
 
         if self._df._select_statement:
-            group_plan = SelectStatement(
-                from_=SelectSnowflakePlan(
-                    snowflake_plan=group_plan, analyzer=self._df._session._analyzer
-                ),
-                analyzer=self._df._session._analyzer,
-            )
+            if isinstance(self._df._select_statement, MockSelectStatement):
+                group_plan = MockSelectStatement(
+                    from_=MockSelectSnowflakePlan(
+                        snowflake_plan=group_plan, analyzer=self._df._session._analyzer
+                    ),
+                    analyzer=self._df._session._analyzer,
+                )
+            else:
+                group_plan = SelectStatement(
+                    from_=SelectSnowflakePlan(
+                        snowflake_plan=group_plan, analyzer=self._df._session._analyzer
+                    ),
+                    analyzer=self._df._session._analyzer,
+                )
```

Comment:
Is it possible we don't change this part and do convert a`SelectSnowflakePlan` to `MockSelectSnowflakePlan` in the mock code?

Reply:
I moved the logic to `mock_analyzer`, I think that's a good place to intercept and re-wrap the plan into a mock one